File: CarRental/ClientApp/src/app/classes/BookingDto.ts
import { CustomerDto } from "./CustomerDto";
import { BranchDto } from "./BranchDto";
import { BookingExtraDto } from "./BookingExtraDto";
import { CarDto } from "./CarDto";

 

    // $Classes/Enums/Interfaces(filter)[template][separator]
    // filter (optional): Matches the name or full name of the current item. * = match any, wrap in [] to match attributes or prefix with : to match interfaces or base classes.
    // template: The template to repeat for each matched item
    // separator (optional): A separator template that is placed between all templates e.g. $Properties[public $name: $Type][, ]

    // More info: http://frhagn.github.io/Typewriter/

    
    export class BookingDto {
        
        // Id
        public Id: number = 0;
        // Customer
        public Customer: CustomerDto = null;
        // FromBranch
        public FromBranch: BranchDto = null;
        // ToBranch
        public ToBranch: BranchDto = null;
        // bookingextras
        public bookingextras: BookingExtraDto[] = [];
        // Car
        public Car: CarDto = null;
        // FromDate
        public FromDate: string = null;
        // ReturnDate
        public ReturnDate: string = null;
        // isActive
        public isActive: string = null;
        // Status
        public Status: string = null;
        // CustomerId
        public CustomerId: number = 0;
        // CarId
        public CarId: number = 0;
        // FromBranchID
        public FromBranchID: number = 0;
        // ToBranchID
        public ToBranchID: number = 0;
        // IsNewCustomer
        public IsNewCustomer: boolean = false;
    }

File: CarRental/Dtos/BookingDto.cs
using CarRental.Helpers;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace CarRental.Dtos
{
    public class BookingDto : AddressDto
    {
        public int Id { get; set; }

        public CustomerDto Customer { get; set; }
        public BranchDto FromBranch { get; set; }
        public BranchDto ToBranch { get; set; }
        public IList<BookingExtraDto> bookingextras { get; set; }
        public CarDto Car { get; set; }

        private string _fromDate;
        private string _returnDate;
        public string FromDate
        {
            get
            {
                return        
                    Utility.ConvertToDatetime(_fromDate);

            }
            set { _fromDate= Utility.ConvertToDatetime(value); }
       
        }
        public string ReturnDate
        {
            get
            {
                return Utility.ConvertToDatetime(_returnDate);

            }
            set
            {

                _returnDate= Utility.ConvertToDatetime(value);
            }
        }
        //  public DateTime ActualReturnDate { get; set; }
        public string isActive { get; set; }
        public string Status {
            get
            {
                return isActive == "True" ? "Active" : "Completed";
                     

            }
            

        }
        public int CustomerId { get; set; }
        public int CarId { get; set; }
        public int FromBranchID { get; set; }
        public int ToBranchID { get; set; }
        public bool IsNewCustomer { get; set; }
        public DateTime CreatedOn { get; set; }
        public string CreatedBy { get; set; }
    }
}
